libexec/apply_catalog.rb

```diff
@@ -62,7 +62,12 @@
 
     transport = Puppet::ResourceApi::Transport.connect(conn_info['type'], transport_conn_info)
     transport_wrapper = Puppet::ResourceApi::Transport::Wrapper.new(conn_info['type'], transport)
-    Puppet::Util::NetworkDevice.instance_variable_set(:@current, transport_wrapper)
+    if  Puppet::Util::NetworkDevice.respond_to?(:set_transport)
+      Puppet::Util::NetworkDevice.set_transport(conn_info['type'], transport_wrapper)
```

Comment:
either this takes a `Device`, then it should be called `set_device`, or if it is called `set_transport`, it should take a `Transport`.

Reply:
hmm it seems to have ended up taking a transport but being called set_device. Oh well.